test/units.rb

```diff
@@ -2,217 +2,217 @@
-	end
-	
-	it "should not equal nil" do
-	    meter.wont_be_nil
-	end
-    end
-
-    describe 'arithmetic with like units' do
-	it 'must exponentiate' do
-	    (Units.meters**2).must_equal Units.new(meters:2)
-	end
-    end
-
-    describe "arithmetic with mixed units" do
-	let(:meter) { Units.new(:meters) }
-
-	it "should multiply" do
-	    (meter * Units.new(:inches)).must_equal Units.new(:meters => 1, :inches => 1)
-	end
-	
-	it "should multiply by nil" do
-	    (meter * nil).must_equal meter
-	end
-	
-	it "should divide by nil" do
-	    (meter / nil).must_equal meter
-	end
-    end
-
-    describe "comparison" do
-	describe "spaceship" do
-	    it "must return 0 for equal units" do
-		(Units.new(:meters) <=> Units.new(:meters)).must_equal 0
-	    end
-
-	    it "must return nil for unequal units" do
-		(Units.new(:meters) <=> Units.new(:inches)).must_be_nil
-	    end
-	end
-    end
-
-    describe "conversion" do
-	it "should have an inspect method" do
-	    Units.new('meters').inspect.must_equal 'meter'
-	end
-	
-	it "should have a to_s method" do
-	    Units.new('meters').to_s.must_equal 'meter'
-	end
-	
-	it "should have a to_abbreviation method" do
-	    Units.new('centimeters').to_abbreviation.must_equal 'cm'
-	end
-
-	it "must have a convert method that converts a value to new units" do
-	    Units.meter.convert(10, 'inch').must_equal 393.701
-	end
-
-	it 'must convert degrees to radians' do
-	    90.degrees.radians.must_be_close_to Math::PI/2
-	end
-    end
-
-    it "must square root" do
-	(Units.new('meters')*Units.new('meters')).square_root.must_equal Units.new('meters')
-    end
-
-    describe 'when checking conversion validity' do
-	it 'must accept inches and meters' do
-	    Units.meter.valid_conversion?('inch').must_equal true
-	    Units.inch.valid_conversion?('meter').must_equal true
-	    Units.meter.valid_conversion?(:inch).must_equal true
-	    Units.inch.valid_conversion?(:meter).must_equal true
-	end
-
-	it 'must accept conversion to self' do
-	    Units.meter.valid_conversion?(:meter).must_equal true
-	    Units.meter.valid_conversion?(:mm).must_equal true
-	end
-
-	it 'must accept mixed prefixes' do
-	    Units.inch.valid_conversion?(:mm).must_equal true
-	end
-
-	it 'must reject nonsense conversions' do
-	    Units.meter.valid_conversion?(:foo).must_equal false
-	end
-
-	it 'must reject invalid conversions' do
-	    Units.meter.valid_conversion?(:hertz).must_equal nil
-	end
-    end
-
-    describe 'when overriding is_a?' do
-	it 'must be a base unit' do
-	    Units.meters.is_a?(:meters).must_equal true
-	end
-
-	it 'must be a prefixed unit' do
-	    Units.mm.is_a?(:meters).must_equal true
-	end
+  describe "when validating strings" do
+    let(:units_s)   { Units::UNITS.map {|u| (u.to_s + 's').to_sym } }
+    let(:units_es)  { Units::UNITS.map {|u| (u.to_s + 'es').to_sym } }
+
+    it "should accept all valid units" do
+      Units::UNITS.each {|unit| _(Units.valid_unit?(unit)).must_equal(true) }
```

Comment:
`_()` may be confused with GetText's syntax. Can we use `assert_*` statements?
https://github.com/ruby-gettext/gettext#_-or-gettext-basic-translation-method